### PR TITLE
Implementation of CardPart Video View

### PR DESCRIPTION
**Pull request by badrinathvm**
## Before you make a Pull Request, read the important guidelines:
Implementation of CardPartVideoView.

Video_iOS

## Issue Link :link:
 <ul>
   <li> New Feature </li>
   <li> Does it break any existing functionality? - None</li>
</ul>

## Goals of this PR :tada:
 <ul>
   <li> Why is the change important?  - Addition of new component to CardParts for displaying Videos</li>
   <li> What does this fix? </li>
   <li> How far has it been tested? - Unit / UI Tested </li>
 </ul>
 
## How Has This Been Tested :mag:

Please let us know if you have tested your PR and if we need to reproduce the issues. Also, please let us know if we need any relevant information for running the tests.

<ul>
 <li> User Interface Testing : YES  </li>
 <li> Application Testing: YES </li>
</ul>

## Test Configuration :space_invader:

<ul>
 <li> Xcode version: 12.0</li>
 <li> Device/Simulator: 11.0 </li>
 <li> iOS version || MacOSX version: Catalina - 10.15.7 </li>
</ul>

## Things to check on :dart:

 - [ ] My Pull Request code follows the coding standards and styles of the project 
 - [ ] I have worked on unit tests and reviewed my code to the best of my ability 
 - [ ] I have used comments to make other coders understand my code better 
 - [ ] My changes are good to go without any warnings 
 - [ ] I have added unit tests both for the happy and sad path 
 - [ ] All of my unit tests pass successfully before pushing the PR 
 - [ ] I have made sure all dependent downstream changes impacted by my PR are working

### CODE REVIEW

File: CardParts/src/Classes/Card Parts/CardPartVideoView.swift [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        let heightConstraint = NSLayoutConstraint(item: self, attribute: .height, relatedBy: .equal, toItem: nil, attribute: .height, multiplier: 1, constant: playerHeight)
+        heightConstraint.identifier = "CPVHeight"
+        self.addConstraint(heightConstraint)
+    }
+    
+    // configures the video for the view controller.
+    public func configureVideo(for viewController: UIViewController) {
+        viewController.view.addSubview(controller.view)
+        viewController.addChild(controller)
+        self.addSubview(controller.view)
```
[the remark this answers is not in the document]
Comment (badrinathvm) [reply CardParts/src/Classes/Card Parts/CardPartVideoView.swift]: Good point. Fixed.

**Comment by rcole34** [comment]
@croossin @bharathmurs this seems to be working fine for Badari, but I wasn't sure if this is the "CardParts" way of adding a subview from another controller? I was thinking he might need to set the `viewController` property of the `CardPartsVideoView` to get this snippet from `setupCardParts` to run, but it seems to work without that also https://github.com/intuit/CardParts/blob/master/CardParts/src/Classes/CardPartsViewController.swift#L85:L88

**Comment by badrinathvm** [comment]
@rcole34  `self.viewContoller` from CardPartView is not being used, the above comment might not be valid anymore.

File: CardParts/src/Classes/Card Parts/CardPartVideoView.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    
+    // url of the video to load on the AVPlayer
+    public var videoUrl:URL
+    
+    // player instance
+    fileprivate var player:AVPlayer!
+    
+    fileprivate let controller = AVPlayerViewController()
+    
+    // convinence initilazer for setting up the videp player
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartVideoView.swift:24]: Typo in video

**Review by croossin — APPROVED** [review]
Looks good to me.

**Comment by croossin** [comment]
Just curious does this support picture-in-picture out of the box

File: CardParts/src/Classes/Card Parts/CardPartVideoView.swift (hunk 2 above)
Comment (badrinathvm) [reply CardParts/src/Classes/Card Parts/CardPartVideoView.swift]: Fixed

**Pull request merged**